luqum/thread.py

```diff
@@ -1,19 +1,19 @@
 import threading
 
-from ply import lex
-
 from . import parser
 
-
 thread_local = threading.local()
 
 
 def parse(input=None, lexer=None, debug=False, tracking=False):
     """A (hopefully) thread safe version of :py:meth:`luqum.parser.parse`
 
-    PLY is not thread safe because of its lexer state, but cloning it we can be thread safe.
-    see: https://github.com/jurismarches/luqum/issues/72
+    PLY is not thread safe because of its lexer state, but cloning it we can be
+    thread safe. see: https://github.com/jurismarches/luqum/issues/72
+
+    Warning: The parameter ``lexer``, ``debug`` and ``tracking`` are not used.
+    They are still present for signature compatibility.
```

Comment:
Is it impossible to keep this parameters ?

(sorry I don't remember why they are not used !)

Reply:
The parameter are kept. I just documented that there are (and were) not used.

![image](https://github.com/user-attachments/assets/5180a580-4df2-4615-b17c-380849353f15)

I don't understand what you are asking for.

Reply:
No, it's just that I did this code, but I don't remember why :grimacing: and as you had a look at it, I was wandering if you know why. But it's perfectly fine.